### DISCUSSION TITLE
Diesel 2.2.2 with 128-column-tables feature hangs on RustRover but works on VSCode

### DISCUSSION THREAD

**Discussion by huangshengling** [diesel-rs/diesel#4163] [Q&A]
I am experiencing an issue where the RustRover IDE hangs during compilation when using Diesel version 2.2.2 with the 128-column-tables feature. The compilation process gets stuck at "Preparing data for name resolution." However, the same setup works fine when using VSCode.
**Steps to Reproduce**
Install RustRover 2024.1.
Configure Diesel version 2.2.2 with the 128-column-tables feature in Cargo.toml.
Attempt to compile the project in RustRover.
Observe that the compilation hangs at the "Preparing data for name resolution" stage.
**Expected Behavior**
The project should compile successfully without hanging.
**Actual Behavior**
The compilation process hangs indefinitely at the "Preparing data for name resolution" stage in RustRover. However, the same project compiles successfully in VSCode.
**Environment**
**RustRover Version**: 2024.1
![1](https://github.com/user-attachments/assets/fe137816-09cd-4fbd-8248-344fcb4b5362)

**Diesel Version**: 2.2.2 with features ["chrono", "sqlite", "128-column-tables"]
![2](https://github.com/user-attachments/assets/d156ac6d-5434-48ed-90c7-0f3eeb3cc80f)

**Compilation Info**:
![3](https://github.com/user-attachments/assets/5949fe5e-5253-4169-ae42-276398b54c00)

**Operating System**: Manjaro 24.0.5 Wynsdey
![4](https://github.com/user-attachments/assets/da74d2e9-674d-4063-b9f6-da09aabb389c)

Kernel: x86_64 Linux 6.4.14-1-MANJARO
Shell: zsh 5.9
CPU: AMD Ryzen 7 7840HS with Radeon 780M Graphics
RAM: 20414MiB / 31370MiB
**Additional Information**
The issue does not occur when using VSCode with the same project setup.
I have tried cleaning the project and rebuilding, but the issue persists in RustRover.
**Steps Taken to Resolve**
Enabled incremental compilation in Cargo.toml.
Used sccache to cache compilation results.
Cleaned the project with cargo clean and rebuilt it.
Verified that the issue is specific to RustRover and does not occur in VSCode.
**Request for Help**
I am seeking assistance to understand why RustRover hangs during compilation and how to resolve this issue. Any configuration changes or troubleshooting steps that could help resolve this would be greatly appreciated.

**Comment by weiznich** [comment]
Given that the configuration works in different other environments: Can you explain why you believe that this is a bug in diesel and not in rustrover?

We consider only compilation errors from rustc as bugs in diesel, so there would need to be a strong indication that we do something that's not part of the language spec to change something on our side here.

**Comment by huangshengling** [comment]
> Given that the configuration works in different other environments: Can you explain why you believe that this is a bug in diesel and not in rustrover?
> 
> We consider only compilation errors from rustc as bugs in diesel, so there would need to be a strong indication that we do something that's not part of the language spec to change something on our side here.

Given that the configuration works fine in other development environments, I would like to ask if there are any RustRover-specific configurations or troubleshooting steps that could help resolve this issue. Specifically, I would like to understand the following:
Are there any known RustRover-specific issues or limitations that might affect the compilation of Diesel features (such as "128-column-tables")?
Are there any recommended RustRover configurations or debugging steps that could further diagnose and resolve this issue?
I greatly appreciate your help and guidance.

**Reply by weiznich** [reply]
I converted this issue to an discussion as it's no bug report. Our discussion forum is the right place to handle this kind of support questions.

That written: As I neither use rust-rover nor the 128 column feature I cannot give you any more advice here. It might be worth tho reach out tho the RustRover team instead.

**Comment by marvin-hansen** [comment]
@huangshengling Please fill a report with JetBrains. I had way too many database bugs and poor performance with all of their IDE's especially GoLand and PyCharm and ultimately switched to [DBeaver](https://dbeaver.io/) for everything database related.

Also, the build and run tool in RustRover is broken beyond repair. I always use a separate terminal to build and run my project.  Please try first to to compile and run in a separate terminal and see if this works.

**Comment by illode** [comment]
@huangshengling I ran into this same issue with the default 32 column tables. I fixed it by deleting the `.idea` and `target` dirs.

As was said, you should file an issue with Jetbrains, not diesel, as it's a problem on their end. I'm starting to feel like their IDEs are getting buggier and less performant as time goes on, which is quite disappointing.
